rfcs/experimental/flow_graph_node_sets/README.md

```diff
@@ -0,0 +1,256 @@
+# Helper Functions for Expressing Flow Graphs
+
+## Introduction
+
+The flow graph API lacks a simple way to create multiple edges with a single call. The result of this
+limitation is verbose, less-readable code.  For example, the follow pciture shows a flow graph where the
+node `input` has three successors.
+
+<img src="graph_with_many_edges.png" width=500>
+
+Without the proposed extensions, three separate calls to `make_edge` are require to connect the
+single `input` node to its three successors: 
+
+```cpp
+    broadcast_node<int> input(g);
+
+    function_node doubler(g, unlimited, [](const int& v) { return 2 * v; });
+    function_node squarer(g, unlimited, [](const int&) { return v * v; });
+    function_node cuber(g, unlimited, [](const int& v) { return v * v * v; });
+
+    make_edge(input, doubler);
+    make_edge(input, squarer);
+    make_edge(input, cuber);
+```
+
+To reduce verbosity and improve readability, additional functions were added to the flow graph API
+to simpify these cases. A shorter and more readable implementation of the previous code snippet that
+uses these extensions is shown below:
+
+```cpp
+    function_node doubler(g, unlimited, [](const int& v) { return 2 * v; });
+    function_node squarer(g, unlimited, [](const int&) { return v * v; });
+    function_node cuber(g, unlimited, [](const int& v) { return v * v * v; });
+
+    broadcast_node input(precedes(doubler, squarer, cuber));
+```
+
+## Proposal
+
+Several helper functions and additional constructors have been added as experimental features.
+These additions simplify expressing dependencies between nodes when building oneTBB flow graphs.
+This experimental feature is described in more detail in the
+[Developer Guide](https://www.intel.com/content/www/us/en/docs/onetbb/developer-guide-api-reference/2022-0/helper-functions-for-expressing-graphs.html)
+
+There are four main parts to the extension:
+
+- make_node_set function template
+- make_edges function template
+- two additional constructors for each flow graph node
+- follows and precedes function templates
+
+### Node Sets
+
+In this proposal, `node_set` is an exposition-only name for the type returned from the
+`make_node_set` function. The `make_node_set` function template creates a set of nodes that
+can be passed as arguments to the `make_edges`, `follows` and `precedes` functions that are
+described in later sections.
+
+```cpp
+    // node_set is an exposition-only name for the type returned from make_node_set function
+
+    template <typename Node, typename... Nodes>
+    /*unspecified*/ make_node_set( Node& node, Nodes&... nodes );
```

Comment:
As far as I see, current `make_edge` function in the general case only allows setting edges between sender and receiver of the same type. Because of this, it seems useless to allow `make_node_set` to accept nodes with different `input_type` or `output_type`. I think we can think about adding reasonable constraints APIs mentioned in this paper.

Reply:
For multioutput or multi port nodes, the set might contain different types at each index, as long as they match